### ISSUE TITLE
CI testing command

### ISSUE THREAD

**Issue by commitay** [Homebrew/homebrew-cask#48383]
Somewhat related to the proposed changes in https://github.com/Homebrew/homebrew-cask/issues/48267.

Our CI scripts have become more complicated in the past year now that we run install/uninstall, `gpg` and "leftovers" checks.

Implementing a ruby CI test command similar to https://github.com/Homebrew/homebrew-test-bot might be useful as we could add more checks and better handle errors.

It should also allow us to readd `travisbuddy` and "tune" it for better error output. e.g  commenting only on errors which should be actionable by the contributor: audit, style, "source not found" and "leftover" issues.

**Comment by reitermarkus** [comment]
I actually have been experimenting on a local branch and integrated all the leakes file checks into `Audit` and added a `--install` option.

**Comment by commitay** [comment]
Can you open a PR or push you changes so we can take a look? My main concern is that audit is already rather complicated. I'd like to break it up into stages with a separate command rather than trying to fit everything into audit.

**Comment by commitay** [comment]
Forgot to include it in the top post but one check I'd like to add is a `postinstall` for pkgs to see if they install an `app`.

**Comment by commitay** [comment]
Found it https://github.com/Homebrew/brew/compare/master...reitermarkus:audit-install

**Comment by commitay** [comment]
Just to elaborate futher on why I'm not keen on adding non-audit checks to audit is we have no control over the test environment. 

Changes made by Travis could potentially break anything at anytime. Also I think the amount of weird errors we encounter and have to work around means we would still need a separate way of _choosing_ what (and when) we would want to run/install/check. Having everything as part of the core `brew cask audit` command means we don't have that ability available as we can't easily add/remove/edit/test the core code.

**Comment by reitermarkus** [comment]
> adding non-audit checks to audit

Right, I hadn't thought about that. Then I think it should be part of `_audit_modified_casks` and not `audit`. And `_audit_modified_casks` should be renamed to `_test_modified_casks` because it would then be extended to do everything that the current Travis script now does, not just `audit`.

**Comment by commitay** [comment]
> Then I think it should be part of `_audit_modified_casks`

Sorry, I actually meant `_audit_modified_casks` when I said `audit`.

**Comment by reitermarkus** [comment]
So where should it be added then?

**Comment by commitay** [comment]
A separate command that lives outside of the `brew` core code and workflow similar to test-bot.

**Comment by commitay** [comment]
Mentioned in https://github.com/Homebrew/homebrew-cask/issues/49200. 

Check for appcasts after the cask has been installed.
